cranelift/codegen/src/machinst/abi_impl.rs

```diff
@@ -787,9 +837,27 @@ fn gen_store_base_offset_multi<M: ABIMachineSpec>(
     ret
 }
 
+fn legalize_signature(sig: &ir::Signature) -> ir::Signature {
```

Comment:
Since this is a single-purpose legalization enforcing the presence of a StructReturn ret if there's an StructReturn input, how about naming it `ensure_struct_return_is_in_out`, or something more targeted and precise?

Reply:
Done!